src/include/distributed/transaction_management.h

```diff
@@ -66,6 +66,9 @@ extern CoordinatedTransactionState CurrentCoordinatedTransactionState;
 /* list of connections that are part of the current coordinated transaction */
 extern dlist_head InProgressTransactions;
 
+/* whether we've been asked to use 2PC (by calling CoordinatedTransactionUse2PC()) */
+extern bool CoordinatedTransactionUses2PC;
```

Comment:
Yeah I'm not too familiar with the Transaction API, but this kind of global state that needs setting and unsetting during methods seems like a bad pattern or a leaky abstraction.